### PR TITLE
Distinct time description setting it apart from the current candles time

### PR DESCRIPTION
**Pull request by kakulukia**
I though it was the same as the current candles time. :(

That came from:
"TIP

Just like in the API of crypto exchanges, and TradingView, each candle's timestamp is the beginning of that time period, not the ending but the actual time it began."

### CODE REVIEW

**Comment by saleh-mir** [comment]
Hey man, thank you for the pull request. Your description would be true if we were only using it for practice, but in live trading, that's not the case anymore. In live trading, It can point to the actual current time. But in backtest we are trying to simulate the current time, so that's why we use the current candle's value.

**Comment by kakulukia** [comment]
Oh well, now im officially confused as 

ipdb> self.time
1704153600000.0
and
ipdb> self.candles[-1][0]
1704067200000.0

are never the same for me. :( 

I consistently get a self.time that is the closing time of the last candle:
image

Tested with multiple time frames. 
image

2h or 15 minutes (or any other time frame i tested) it always matches up.

Is there something wrong with my installation of Jesse? 
But actually that totally makes sense, because as you say in live mode we will only get the last candle after its finished.

**Comment by saleh-mir** [comment]
> But actually that totally makes sense, because as you say in live mode we will only get the last candle after its finished.
Yes, there is a scenario where the current time in live trading may differ from the closing timestamp of candles. For instance, when an order is executed and triggers the on_open_position() function, the current time in your code may not match the closing timestamp of the current candle.

**Comment by kakulukia** [comment]
Understood, but what i was trying to say:
 "But in backtest we are trying to simulate the current time, so that's why we use the current candle's value." 
That is not true.

In other words: 
1. backtest - self.time is always the closing time of the current candle
2. live - self.time will mostly be the closing time of the current candle - exceptions are on_open_position(), ...

**Comment by kakulukia** [comment]
Maybe that changed some time in the past and self.time for backtesting actually was the same as self.candles[-1][0]?

**Comment by saleh-mir** [comment]
Okay, my apologies. When I refer to the timestamp of the current candle, I mean we utilized it, though not precisely that. Also, when I mention the current candle, I am not specifically referring to the trading candle but rather the one-minute candle. Therefore, the calculation would be as follows:
time = current_1m_candle_timestamp + 60000.

The one-minute candle is relevant because the scenario I described with orders being executed applies to backtest as well. However, in live trading, this could occur at any moment, whereas in backtests, the shortest time frame available is one minute.

**Comment by kakulukia** [comment]
Okay, let me rephrase the above commit tomorrow also considering the current conversation. I totally get you point! But as you just confirmed it's never going to be the current candles opening time either. So the current description is misleading. And sure as hell confused me.

**Comment by saleh-mir** [comment]
You're right. it needs to be modified based on the fact that now it can be used in both backtests and live trading.

**Comment by kakulukia** [comment]
image

This is my new proposal.

**Comment by saleh-mir** [comment]
I think the first paragraph is great, but I don't think we need the two bullet points.

**Comment by kakulukia** [comment]
I get what you are thinking. It's a bit to much. Those two bullet points implicitly tell the same story. But even you were confused at the start of this PR!
So while I'm 50/50 on this and will gladly remove em, let me share my thoughts that got me to put them there in the first place: This way it's crystal clear, its the gist of our conversation, reinforces the understanding of the existing event system, Google likes links (well nobody really knows, but thats what people claim to be important for SEO) and anybody who might have also thought "so that's why we use the current candle's value" will for sure unlearn that while reading that new description.

Your call: with or without?

**Comment by saleh-mir** [comment]
Without because I want the user to just use it, they don't really need to know how it's been calculated unless they are curious.

The first bullet point is misleading because it only shows the closing time of the current candle if it's being executed regularly. However, if it's being executed at other times, it would be different. In live trading, we simply return the current timestamp and don't care about the candle at all. Explaining how it's simulated in backtest would only add more confusion. Users who are curious about such things can check out the source code and see it for themselves.
